### PR TITLE
Port BackgroundJob admin settings to vue

### PR DESCRIPTION
**Pull request by CarlSchwan**
image

### CODE REVIEW

File: apps/settings/lib/Settings/Admin/Server.php [hunk 1]
```diff
@@ -63,27 +60,30 @@ public function __construct(IDBConnection $connection,
...
-			'lastcron' => $this->config->getAppValue('core', 'lastcron', false),
-			'cronMaxAge' => $this->cronMaxAge(),
-			'cronErrors' => $this->config->getAppValue('core', 'cronErrors'),
-			'cli_based_cron_possible' => function_exists('posix_getpwuid'),
-			'cli_based_cron_user' => function_exists('posix_getpwuid') ? posix_getpwuid(fileowner(\OC::$configDir . 'config.php'))['name'] : '',
-			'profileEnabledGlobally' => $this->profileManager->isProfileEnabled(),
-		];
+		// Background jobs
+		$this->initialStateService->provideInitialState('backgroundJobsMode', $this->config->getAppValue('core', 'backgroundjobs_mode', 'ajax'));
+		$this->initialStateService->provideInitialState('lastCron', (int)$this->config->getAppValue('core', 'lastcron', 0));
```
Comment (CarlSchwan) [inline apps/settings/lib/Settings/Admin/Server.php:72]:
```suggestion
		$this->initialStateService->provideInitialState('lastCron', (int)$this->config->getAppValue('core', 'lastcron', '0'));
```

File: apps/settings/src/components/BasicSettings/BackgroundJob.vue [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+	- GNU Affero General Public License for more details.
+	-
+	- You should have received a copy of the GNU Affero General Public License
+	- along with this program. If not, see <http://www.gnu.org/licenses/>.
+	-
+-->
+
+<template>
+	<SettingsSection
+		:title="t('settings', 'Background jobs')"
```
Comment (CarlSchwan) [inline apps/settings/src/components/BasicSettings/BackgroundJob.vue:25]:
```suggestion
	<SettingsSection :title="t('settings', 'Background jobs')"
```

File: apps/settings/src/components/BasicSettings/BackgroundJob.vue [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+const cliBasedCronUser = loadState('settings', 'cliBasedCronUser', 'www-data')
+const backgroundJobsDocUrl = loadState('settings', 'backgroundJobsDocUrl')
+
+export default {
+	name: 'BackgroundJob',
+
+	components: {
+		CheckboxRadioSwitch,
+		SettingsSection,
+		ActionText,
```
Comment (CarlSchwan) [inline apps/settings/src/components/BasicSettings/BackgroundJob.vue:109]:
```suggestion
```

File: apps/settings/src/components/BasicSettings/BackgroundJob.vue [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+		</em>
+	</SettingsSection>
+</template>
+
+<script>
+import { loadState } from '@nextcloud/initial-state'
+import { showError } from '@nextcloud/dialogs'
+import CheckboxRadioSwitch from '@nextcloud/vue/dist/Components/CheckboxRadioSwitch'
+import SettingsSection from '@nextcloud/vue/dist/Components/SettingsSection'
+import ActionText from '@nextcloud/vue/dist/Components/ActionText'
```
Comment (CarlSchwan) [inline apps/settings/src/components/BasicSettings/BackgroundJob.vue:90]:
```suggestion
```

File: apps/settings/src/components/BasicSettings/BackgroundJob.vue [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+			</span>
+
+			<span class="warning" v-else-if="Date.now() / 1000 - cronMaxAge > 12 * 3600 && backgroundJobsMode !== 'cron'">
+				{{ t('settings', "Some jobs haven’t been executed since {maxAgeRelativeTime}. Please consider switching to system cron.", {maxAgeRelativeTime}) }}
+			</span>
+
+			<span v-else>
+				{{ t('settings', 'Last job ran {relativeTime}.', {relativeTime}) }}
+			</span>
+		</template>
```
Comment (skjnldsv) [inline apps/settings/src/components/BasicSettings/BackgroundJob.vue:44]: Would be cleaner as a computed :)

File: apps/settings/src/main-admin-basic-settings.js [hunk 6]
```diff
@@ -43,6 +44,9 @@ Vue.mixin({
 	},
 })
 
+const BackgroundJobView = Vue.extend(BackgroundJob)
+new BackgroundJobView().$mount('.vue-admin-background-job')
```
Comment (skjnldsv) [inline apps/settings/src/main-admin-basic-settings.js:48]: We usually use an ID to ensure uniqueness

Comment (skjnldsv) [reply apps/settings/src/main-admin-basic-settings.js:48]: But I also see below someone used a class, if you want to change both :shrug:

File: apps/settings/src/components/BasicSettings/BackgroundJob.vue (hunk 5 above)
Comment (CarlSchwan) [reply apps/settings/src/components/BasicSettings/BackgroundJob.vue]: done :)

File: apps/settings/src/components/BasicSettings/BackgroundJob.vue [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+		<span class="error" v-else>
+			{{ t('settings', 'Background job didn’t run yet!') }}
+		</span>
+
+		<fieldset>
+		<CheckboxRadioSwitch type="radio"
+			:checked.sync="backgroundJobsMode"
+			value="ajax"
```
Comment (CarlSchwan) [inline apps/settings/src/components/BasicSettings/BackgroundJob.vue:53]: TODO: need name attribute

File: apps/settings/src/components/BasicSettings/BackgroundJob.vue [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+			<span v-else>
+				{{ t('settings', 'Last job ran {relativeTime}.', {relativeTime}) }}
+			</span>
+		</template>
+
+		<span class="error" v-else>
+			{{ t('settings', 'Background job didn’t run yet!') }}
+		</span>
+
+		<fieldset>
```
Comment (CarlSchwan) [inline apps/settings/src/components/BasicSettings/BackgroundJob.vue:50]:
```suggestion
```

File: apps/settings/src/components/BasicSettings/BackgroundJob.vue [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+			{{ t('settings', 'Cron (Recommended)') }}
+		</CheckboxRadioSwitch>
+		<em v-if="cliBasedCronPossible">{{ cronLabel }}</em>
+		<em v-else>
+			{{ t('settings', 'To run this you need the PHP POSIX extension. See {linkstart}PHP documentation{linkend} for more details.', {
+				linkstart: '<a href="https://www.php.net/manual/en/book.posix.php">',
+				linkend: '</a>',
+			}) }}
+		</em>
+		</fieldset>
```
Comment (CarlSchwan) [inline apps/settings/src/components/BasicSettings/BackgroundJob.vue:82]:
```suggestion
```

File: apps/settings/src/components/BasicSettings/BackgroundJob.vue [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+	- GNU Affero General Public License for more details.
+	-
+	- You should have received a copy of the GNU Affero General Public License
+	- along with this program. If not, see <http://www.gnu.org/licenses/>.
+	-
+-->
+
+<template>
+	<SettingsSection :title="t('settings', 'Background jobs')"
+		:description="t('settings', 'For the server to work properly, it\'s important to configure background jobs correctly. Cron is the recommended setting. Please see the documentation for more information.')"
```
Comment (Pytal) [inline apps/settings/src/components/BasicSettings/BackgroundJob.vue:25]: Could we wrap this with backticks so the apostrophe doesn't have to be escaped?

**Review by skjnldsv — APPROVED** [review]

**Review by Pytal — APPROVED** [review]

**Pull request merged**